core/controllers/pages_test.py

```diff
@@ -15,15 +15,148 @@
+        """Test for contact page."""
+        response = self.testapp.get('/contact')
+        self.assertEqual(response.status_int, 200)
+        self.assertEqual(response.content_type, 'text/html')
+        response.mustcontain('I18N_CONTACT_PAGE_HEADING')
+
+
+class DonatePageTest(test_utils.GenericTestBase):
+
+    def test_donate_page(self):
+        """Test for donate page."""
+        response = self.testapp.get('/donate')
+        self.assertEqual(response.status_int, 200)
+        self.assertEqual(response.content_type, 'text/html')
+
+
+class ThanksPageTest(test_utils.GenericTestBase):
+
+    def test_thanks_page(self):
+        """Test for thanks page."""
+        response = self.testapp.get('/thanks')
+        self.assertEqual(response.status_int, 200)
+        self.assertEqual(response.content_type, 'text/html')
+
+
+class ForumPageTest(test_utils.GenericTestBase):
+
+    def test_forum_page(self):
+        """Test for forum page."""
+        response = self.testapp.get('/forum')
+        self.assertEqual(response.status_int, 200)
+        self.assertEqual(response.content_type, 'text/html')
+
+
+class TermsPageTest(test_utils.GenericTestBase):
+
+    def test_terms_page(self):
+        """Test for terms page."""
+        response = self.testapp.get('/terms')
+        self.assertEqual(response.status_int, 200)
+        self.assertEqual(response.content_type, 'text/html')
+
+
+class PrivacyPageTest(test_utils.GenericTestBase):
+
+    def test_privacy_page(self):
+        """Test for privacy page."""
+        response = self.testapp.get('/privacy')
+        self.assertEqual(response.status_int, 200)
+        self.assertEqual(response.content_type, 'text/html')
+
+
+class AboutRedirectPageTest(test_utils.GenericTestBase):
+
+    def test_about_redirect_page(self):
+        """Test for about redirect page."""
+        response = self.testapp.get('/credits')
+        self.assertEqual(response.status_int, 302)
+
+
+class FoundationRedirectPageTest(test_utils.GenericTestBase):
+
+    def test_foundation_redirect_page(self):
+        """Test for foundation redirect page."""
+        response = self.testapp.get('/foundation')
+        self.assertEqual(response.status_int, 302)
+
+
+class TeachRedirectPageTest(test_utils.GenericTestBase):
+
+    def test_teach_redirect_page(self):
+        """Test for teach redirect page."""
+        response = self.testapp.get('/participate')
+        self.assertEqual(response.status_int, 302)
+
+
+class ConsoleErrorPageTest(test_utils.GenericTestBase):
+
+    def test_console_error_page(self):
+        """Test for console error page."""
+        response = self.testapp.get('/console_errors')
+        self.assertEqual(response.status_int, 200)
+
+
+class MaintenancePageTest(test_utils.GenericTestBase):
+
+    def test_maintenance_page(self):
+        """Test for maintenance page."""
+        feconf.ENABLE_MAINTENANCE_MODE = True
+        with self.swap(feconf, 'ENABLE_MAINTENANCE_MODE', True):
```

Comment:
If you are using swap then why are you explicitly changing ENABLE_MAINTENANCE_MODE value on the line above? Ditto everywhere.